chain/network/src/stats/metrics.rs

```diff
@@ -353,6 +366,18 @@ pub(crate) fn record_routed_msg_latency(clock: &time::Clock, msg: &RoutedMessage
     }
 }
 
+// The routed message reached its destination. If the number of hops is known, then update the
+// corresponding metric.
+fn record_routed_msg_hops(msg: &RoutedMessageV2) {
+    // We assume that the number of hops is small.
+    // As long as the number of hops is below 10, this metric will not consume too much memory.
+    if let Some(num_hops) = msg.num_hops {
+        NETWORK_ROUTED_MSG_NUM_HOPS
+            .with_label_values(&[msg.body_variant(), &num_hops.to_string()])
```

Comment:
for safety --  I'd cap num_hops between 0 and 20.

(so that a malicious actor cannot blowup our metrics sizes by sending a lot of malicious messages with different number of hops).

Reply:
Good idea, thanks